apps/dashboard/app/controllers/scripts_controller.rb

```diff
@@ -25,11 +22,20 @@ def create
     end
   end
 
+  # GET   /projects/:project_id/scripts/:id/edit
+  # edit
+  def edit
+  end
+
+  # POST   /projects/:project_id/scripts/:id/save
+  # save the script after editing
+  def save
+    redirect_to project_path(params[:project_id]), notice: 'TODO'
+  end
+
   # POST   /projects/:project_id/scripts/:id/submit
   # submit the job
   def submit
-    project = Project.find(params[:project_id])
-    @script = Script.find(params[:id], project.directory)
     opts = submit_script_params[:script].to_h.symbolize_keys
```

Comment:
Did you want the `find_script` to replace these, or not worry about this for now?

Reply:
Yea I figure we can use `before_action` to get rid of some duplication.